.github/workflows/lint.yml

```diff
@@ -15,5 +15,39 @@ jobs:
         run: pip install jupyterlab~=4.0
       - name: Install JS dependencies
         run: jlpm
-      - name: Run JS Lint
+      - name: Lint TypeScript source
         run: jlpm lerna run lint:check
+
+  lint_py_imports:
+    name: Lint Python imports
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Echo environment details
+        run: |
+          which python
+          which pip
+          python --version
+          pip --version
+
+      # see #546 for context on why this is necessary
+      - name: Create venv
+        run: |
+          python -m venv lint_py_imports
+
+      - name: Install job dependencies
+        run: |
+          source ./lint_py_imports/bin/activate
+          pip install jupyterlab~=4.0
```

Comment:
4.1 is coming out soon, and Jupyter AI is compatible with it. Should we only lock the major version?

Reply:
That's exactly what is already happening here. When a patch version is not provided after `~=`, then `pip` will resolve that to the latest semver-compatible version with the one you provided. That means once 4.1 is released on PyPI, this script will use 4.1.